File: Java/src/ContainerWithMostWater.java
public class ContainerWithMostWater {
    public int maxArea(int[] height) {
        int l = 0, r = height.length - 1, m = (r - l) * Math.min(height[l], height[r]);
        while (l < r) {
            if (height[l] < height[r]) {
                l++;
            } else {
                r--;
            }
            m = Math.max(m, (r - l) * Math.min(height[l], height[r]));
        }
        return m;
    }

    public int oldMaxArea(int[] height) {
        int m = 0;
        for (int i = 0; i < height.length - 1; i++) {
            for (int j = i + 1; j < height.length; j++) {
                int v = (j - i) * Math.min(height[i], height[j]);
                if (v > m) {
                    m = v;
                }
            }
        }
        return m;
    }
}

File: Python3/11.ContainerWithMostWater/ContainerWithMostWater.py
from typing import List


class Solution:
    def maxArea(self, height: List[int]) -> int:
        i = 0
        j = len(height) - 1
        m = 0
        while i < j:
            h = min(height[i], height[j])
            m = max(m, (j - i) * h)
            while i < j and height[i] <= h:
                i += 1
            while i < j and height[j] <= h:
                j -= 1
        return m